service/src/main/kotlin/app/cash/backfila/client/interceptors/OkHttpClientSpecifiedHeadersInterceptor.kt

```diff
@@ -0,0 +1,26 @@
+package app.cash.backfila.client.interceptors
+
+import app.cash.backfila.client.HttpHeader
+import okhttp3.Interceptor
+import okhttp3.Response
+
+internal class OkHttpClientSpecifiedHeadersInterceptor internal constructor(
+  private val headers: List<HttpHeader>,
+) : Interceptor {
+
+  override fun intercept(chain: Interceptor.Chain): Response {
+    if (headers.isEmpty()) {
+      return chain.proceed(chain.request())
+    }
+
+    val requestBuilder = chain.request().newBuilder()
+    for (header in headers) {
+      requestBuilder.header(header.name, header.value)
+    }
+    return chain.proceed(requestBuilder.build())
+  }
+
+  companion object {
+    const val MAX_ALLOWED_HEADERS_SIZE = 2048
```

Comment:
Maybe the size function should be a companion of this?

Reply:
Done